lib/smart_proxy_remote_execution_ssh/api.rb

```diff
@@ -4,6 +4,98 @@ class Api < ::Sinatra::Base
       get "/pubkey" do
         File.read(Ssh.public_key_file)
       end
+      post "/session" do
+        if env["HTTP_CONNECTION"] != "upgrade" or env["HTTP_UPGRADE"] != "raw"
+          return [ 401, "Invalid request: /ssh/session requires connection upgrade to 'raw'" ]
+        end
+        socket = nil
+        if env['rack.hijack?']
+          env['rack.hijack'].call
+          socket = env['rack.hijack_io']
+        end
+        if !socket
+          return [ 501, "Internal error: request hijacking not available" ]
+        end
+
+        params = MultiJson.load(env["rack.input"].read)
+        key_file = ForemanRemoteExecutionCore.settings.fetch(:ssh_identity_key_file)
```

Comment:
In a production setup, `ForemanRemoteExecutionCore` is not available in the proxy, I think.

Reply:
I guess we can use `Proxy::RemoteExecution::Ssh.private_key_file` from https://github.com/theforeman/smart_proxy_remote_execution_ssh/blob/master/lib/smart_proxy_remote_execution_ssh.rb#L24

Reply:
Thanks! Done.